core/sr-primitives/src/lib.rs

```diff
@@ -374,6 +537,127 @@ impl From<codec::Compact<Perbill>> for Perbill {
 	}
 }
 
+/// Representation of a sign bit.
+///
+/// Note that the ordering matters for proper `Ord` derive.
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(PartialEq, Eq, PartialOrd, Ord)]
+pub enum Sign {
+	/// Negative
+	N,
+	/// Positive
+	P,
+}
+
+/// A fixed point number in the range of `[-(u64::max+1), u64::max+1]`. Composed of one `u64`
+///  _natural_ part and a `Perbill` _decimal_ part.
+///
+/// Note that a positive instance is always greater than a negative instance, even if both have `0`
+///  as their natural and decimal part.
```

Comment:
hmm should we prefer it to be derived by hand so positive 0 == negative 0 ?

Reply:
I think he has taken it from floats, there are `-0` != `+0` AFAIK

Reply:
hmm not sure [play.rustlang](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=fn%20main()%7B%0A%20%20%20%20println!(%22%7B%3A%3F%7D%22%2C%20-0.0%20%3C%200.0)%3B%0A%7D)

but I can agree on this actually we can think Positive0 is between 0 and 0+one_per_billion and negative 0 is between 0 and 0-one_per_billion

Reply:
I think @thiolliere is right and we should provide a manual implementation for `PartialOrd` and `Eq` to follow the standard behavior of a float.